File: 4-estruturas-e-bibliotecas/1110.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(void) {
   int n, i, t;
   vector<int> v;

   while (cin >> n and n) {
      for (i = 1; i <= n; ++i) v.insert(v.begin(), i);
      cout << "Discarded cards: ";
      while (v.size() > 1) {
         cout << v.back();
         v.pop_back();
         v.insert(v.begin(), v.back());
         v.pop_back();
         if (v.size() != 1) cout << ", ";
      }
      cout << endl << "Remaining card: " << v.back() << endl;
      v.clear();
   }
   return 0;
}

File: 4-estruturas-e-bibliotecas/1110.py
while True:
    n = int(input())
    if n == 0: break
    v = [x for x in range(1, n+1)]
    print('Discarded cards:', end=' ')
    while len(v) > 1:
        print(v[0], end='')
        del v[0]
        v.append(v[0])
        del v[0]
        if len(v) > 1: print(end=', ')
    print()
    print('Remaining card:', v[0])
